recipes-ni/ni-base-system-image-tests/files/test_fs_permissions.sh

```diff
@@ -2,9 +2,10 @@
 
 source $(dirname "$0")/ptest-format.sh
 
-ptest_change_test $(basename "$0" ".sh") "" "Diff filesystem permissions with previous"
+ptest_change_test $(basename "$0" ".sh") "" "Check known filesystem permissions and diff others with previous"
 
 source /home/admin/.mongodb.creds
+python3 fs_permissions_known.py
```

Comment:
Looks like we aren't using the exit code of this test, so I suspect it won't cause ptest failure. Looking at `ptest-format.sh` it looks like this should be in its own script so that it can have its own `ptest_change_test`, `ptest_pass`, `ptest_fail`, and `ptest_report` calls. Either that, or this one script should separate the two tests so each can produce an independent report.

Reply:
We can OR the results:

```
python3 test_a ...; A_RES=$?
python3 test_b ...; B_RES=$?

if [ $(($A_RES | $B_RES)) -eq 0 ]; then
    ptest_pass
else
    ptest_fail
fi

ptest_report
```

Reply:
@jpautler, is there any preference with regard to these tests being combined like this, where they cannot fail independently?

Reply:
I'm inclined to say we should separate them. Since the test logic and reporting format are different, it makes sense to report the results separately. Also, as I've seen with the dmesg_diff test, there is the potential for diff-based tests to be noisier than average. Hopefully we've thought this file-permissions-diff-test through well-enough that it won't be bad, but I still think it's a good idea to separate the test reporting, just in case. In summary, I see benefits and no downsides to separating them.

Reply:
Would we need to make a whole new package for that, or could it be something as simple as:

```
source $(dirname "$0")/ptest-format.sh

# Diff Permissions Test

ptest_change_test $(basename "$0" ".sh") "" "Diff filesystem permissions with previous"
source /home/admin/.mongodb.creds
python3 fs_permissions_diff.py \
    --server $MONGO_SERVER --user $MONGO_USER --password $MONGO_PASSWORD
if [ $? -eq 0 ]; then
   ptest_pass
else
   ptest_fail
fi
ptest_report

# Known Permissions Test:

ptest_change_test $(basename "$0" ".sh") "" "Check known filesystem permissions"
python3 fs_permissions_known.py
if [ $? -eq 0 ]; then
   ptest_pass
else
   ptest_fail
fi
ptest_report
```

Reply:
There's no need for a new package. This is why the `run-ptest` script exists: to invoke multiple tests that each report their results. See the [kernel-tests package](https://github.com/ni/meta-nilrt/blob/nilrt/master/hardknott/recipes-kernel/kernel-tests/kernel-tests-files/run-ptest) for an example.